opm/grid/cpgrid/Geometry.hpp

```diff
@@ -505,8 +528,16 @@ namespace Dune
             /// @brief Get the cor-th of 8 corners of the hexahedral base cell.
             GlobalCoordinate corner(int cor) const
             {
-                assert(allcorners_ && cor_idx_);
-                return (allcorners_->data())[cor_idx_[cor]].center();
+                return std::visit(
+                    Dune::overload(
```

Comment:
I just noticed that there exists `Opm::VisitorOverloadSet` too. Is there a reason for choosing one or the other?